mlir/lib/Conversion/VectorToLLVM/ConvertVectorToLLVM.cpp

```diff
@@ -532,9 +532,19 @@ static Value createVectorLengthValue(ConversionPatternRewriter &rewriter,
   auto vShape = vType.getShape();
   assert(vShape.size() == 1 && "Unexpected multi-dim vector type");
 
-  return rewriter.create<LLVM::ConstantOp>(
+  Value vLen = rewriter.create<LLVM::ConstantOp>(
       loc, rewriter.getI32Type(),
       rewriter.getIntegerAttr(rewriter.getI32Type(), vShape[0]));
+
+  if (!vType.getScalableDims()[0])
+    return vLen;
+
+  // Create VScale*vShape[0] and return it as vector length.
```

Comment:
[nit] We tend to write "vscale" rather than VScale. Also, why `Shape` rather than `shape[0]` or (even better, referring to a C++ variable): `vShape[0]`. In fact, you could rename `vLen` as `baseVecLength` to make the variable names more descriptive and use that in the comment.

Reply:
Changed to `vScale * baseVecLength`, refering to actual variable names in code.

Reply:
I feel duplicating all tests is a bit redundant.. the triton test case is using 'add', other vector.mask %mask {vector.reduction <op> ...} tests don't offer additional coverage on code path in mlir/lib/Conversion/VectorToLLVM/ConvertVectorToLLVM.cpp, i.e., the part maps vector.reduction <op> to llvm.intr.vp.reduce.<op> is not changed. I've dup-ed some tests